cvat-data/src/js/cvat-data.js

```diff
@@ -116,56 +119,96 @@ class FrameProvider {
         if (frameNumber in this._frames) {
             return this._frames[frameNumber];
         }
-
         return null;
     }
 
+
     /*
-        Method decodes a block of data
+        Method start asynchronic decode a block of data
 
         @param block - is a data from a server as is (ts file or archive)
         @param start {number} - is the first frame of a block
         @param end {number} - is the last frame of a block + 1
+        @param needed_frame {number} - number of frame that should be returned as soon as
+        @param callback - callback)
+
     */
-    decode(block, start, end) {
-        async function decode(resolve, reject) {
-            // no more requests possible during one is running
+
+    start_decode(block, start, end, needed_frame, callback)
```

Comment:
- js preferred style is camelCase
- I believe we don't need the argument "needed_frame" here. Just pass a number of decoded frame to the callback and the callback should decide itself what to do with it.